Tribler/community/gigachannel/community.py

```diff
@@ -92,13 +111,67 @@ def on_blob(self, peer, blob):
         if reply_blob:
             self.endpoint.send(peer.address, self.ezr_pack(self.NEWS_PUSH_MESSAGE, RawBlobPayload(reply_blob)))
 
+    # def create_search_request(self, query_filter, metadata_type='', sort_by="HEALTH", sort_asc=0, hide_xxx=True):
+    #     cache = self.request_cache.add(SearchRequest(self.request_cache, query_filter))
+    #     self._logger.info("Sending search query:%s, identifier:%s", query_filter, cache.number)
+    #
+    #     return SearchRequestPayload(cache.number, query_filter, metadata_type, sort_by, sort_asc, hide_xxx)
+
+    def send_search_request(self, query_filter, metadata_type='', sort_by="HEALTH", sort_asc=0, hide_xxx=True):
+        self.search_counter += 1
+        request = SearchRequestPayload(self.search_counter, query_filter, metadata_type, sort_by, sort_asc, hide_xxx)
+        self._logger.debug("Started remote search for query:%s", query_filter)
+
+        for peer in self.get_peers()[:max_search_peers]:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_REQUEST, request))
+
+    @lazy_wrapper(SearchRequestPayload)
+    def on_search_request(self, peer, request):
+        unicode_query_filter = cast_to_unicode_utf8(request.query_filter)
```

Comment:
Please test the mechanism with UTF-8 characters, i.e. by doing this in the unit test.

Reply:
Also, whoa, being able to SQL inject attack your own database wasn't that big of a deal. But now **you're processing raw queries from third parties**.

Reply:
@devos50 Will update the test for UTF-8 characters.
@qstokkink query filter is a search query from the GUI which is processed by Pony ORM, SQL injection is not possible. It is not a SQL query to be processed as it is.

Reply:
Had a bit of an in-depth look into how Pony/FTS5 structures queries:

Queries you get from third parties **are parsed** using the FTS5 grammar, but **queries do not interact with the SQL data**. In other words, you can still force someone to parse horrible queries, but you can't actually modify their data. For example, something like this is a perfectly legal query:

```python
u'(((("a"*)OR("b"*))AND(("c"*)OR("d"*))OR(("e"*)OR("f"*))OR(("g"*)OR("h"*))OR(("i"*)OR("j"*))OR(("k"*)OR("l"*))OR(("m"*)OR("n"*))OR(("o"*)OR("p"*))) NOT (((^"q"*)OR(^"r"*))OR((^"s"*)OR(^"t"*))OR((^"u"*)OR(^"v"*))OR((^"w"*)OR(^"x"*))OR((^"y"*)OR(^"z"*))OR((^"a"*)OR(^"b"*))OR((^"c"*)OR(^"d"*))OR((^"e"*)OR(^"f"*)))) OR (((("b"*)OR("c"*))AND(("d"*)OR("e"*))OR(("f"*)OR("g"*))OR(("h"*)OR("i"*))OR(("j"*)OR("k"*))OR(("l"*)OR("m"*))OR(("n"*)OR("o"*))OR(("p"*)OR("q"*))) NOT (((^"r"*)OR(^"s"*))OR((^"t"*)OR(^"u"*))OR((^"v"*)OR(^"w"*))OR((^"x"*)OR(^"y"*))OR((^"z"*)OR(^"a"*))OR((^"b"*)OR(^"c"*))OR((^"d"*)OR(^"e"*))OR((^"f"*)OR(^"g"*))))'
```

**tl;dr** this allows you to be a jerk to others, but it's not a security risk. This is fine for now.